testcode.c

```diff
@@ -0,0 +1,34 @@
+#include <assert.h>
+#include "CurrentMeasurementRange.h"
+
+int main()
+{
+//TDD Step 1: Write test code
+printf("Test case 1");
+int samplesarray[] = {4,5};
+int arraysize = sizeof(samplesarray) / sizeof(samplesarray[0]);
+assert(DetectRangecountandPrintdetails(samplesarray,arraysize) == 1);
+//End of step 1 error observed as expected : error: ‘DetectContinuousRanges’ was not declared in this scope
+printf("\nTest case 2");
+int samplesarray1[] = {3, 3, 5, 4, 10, 11, 12};
+int arraysize1 = sizeof(samplesarray1) / sizeof(samplesarray1[0]);
+assert(DetectRangecountandPrintdetails(samplesarray1,arraysize1) == 2);
+printf("\nTest case 3");
+int samplesarray2[] = {3, 30, 5, 7, 10, 15, 12, 20};
+int arraysize2 = sizeof(samplesarray2) / sizeof(samplesarray2[0]);
+assert(DetectRangecountandPrintdetails(samplesarray2,arraysize2) == 0);
+printf("\nTest case 4");
+int samplesarray3[] = {-3, -30, -5, -7, -10, -11, -12, 20, -6};
+int arraysize3 = sizeof(samplesarray3) / sizeof(samplesarray3[0]);
+assert(DetectRangecountandPrintdetails(samplesarray3,arraysize3) == 2);
```

Comment:
There are no asserts for the start and end of the ranges. This means the output needs to be checked manually. Manual checks are time-consuming, error-prone and more importantly, require familiarity to know if it's correct or not. A new developer would find it hard to detect errors.
One way out - would be to test the detection of the ranges separately. And then print them.

Reply:
I have added test cases to check start and end of range and number of readings also